marlowe-playground-client/src/JavascriptEditor/Types.purs

```diff
@@ -1,10 +1,70 @@
 module JavascriptEditor.Types where
 
+import Prelude
+import Analytics (class IsEvent, Event)
+import Analytics as A
+import Data.Either (Either(..))
+import Data.Lens (Lens', Prism', prism)
+import Data.Lens.Record (prop)
+import Data.Maybe (Maybe(..))
+import Data.Symbol (SProxy(..))
+import Halogen.Monaco (KeyBindings(..))
+import Halogen.Monaco as Monaco
 import Language.Javascript.Interpreter as JS
 import Marlowe.Semantics (Contract)
 
-data JSCompilationState
-  = JSNotCompiled
-  | JSCompiling
-  | JSCompilationError JS.CompilationError
-  | JSCompiledSuccessfully (JS.InterpreterResult Contract)
+data CompilationState
+  = NotCompiled
+  | Compiling
+  | CompilationError JS.CompilationError
+  | CompiledSuccessfully (JS.InterpreterResult Contract)
+
+_CompiledSuccessfully :: Prism' CompilationState (JS.InterpreterResult Contract)
+_CompiledSuccessfully = prism CompiledSuccessfully unwrap
+  where
+  unwrap (CompiledSuccessfully x) = Right x
+
+  unwrap y = Left y
```

Comment:
That's a lovely prism you have there Father.

![Mrs Doyle](https://vignette.wikia.nocookie.net/fatherted/images/c/c1/Mrs_Doyle.jpg/revision/latest/top-crop/width/360/height/450?cb=20160110234406)

Reply:
to be fair I did just copy and paste it